diff --git a/lietorch/include/dispatch.h b/lietorch/include/dispatch.h
index 9c010a8..4efe843 100644
--- a/lietorch/include/dispatch.h
+++ b/lietorch/include/dispatch.h
@@ -36,7 +36,7 @@
   [&] {                                                                              \
     const auto& the_type = TYPE;                                                     \
     /* don't use TYPE again in case it is an expensive or side-effect op */          \
-    at::ScalarType _st = ::detail::scalar_type(the_type);                            \
+    at::ScalarType _st = the_type.scalarType();                            \
     switch (_st) {                                                                   \
       PRIVATE_CASE_TYPE(GROUP_INDEX, at::ScalarType::Double, double, __VA_ARGS__)    \
       PRIVATE_CASE_TYPE(GROUP_INDEX, at::ScalarType::Float, float, __VA_ARGS__)      \

